tests/stubs/cloud/controllerstub/podmanager.go

```diff
@@ -0,0 +1,158 @@
+package controllerstub
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/kubeedge/beehive/pkg/common/log"
+	"github.com/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/kubeedge/tests/stubs/common/constants"
+	"github.com/kubeedge/kubeedge/tests/stubs/common/types"
+	"github.com/kubeedge/kubeedge/tests/stubs/common/utils"
+)
+
+// NewPodManager creates pod manger
+func NewPodManager() (*PodManager, error) {
+	event := make(chan *model.Message)
+	pm := &PodManager{event: event}
+	return pm, nil
+}
+
+// PodManager is a manager watch pod change event
+type PodManager struct {
+	// event
+	event chan *model.Message
+	// pods map
+	pods sync.Map
+}
+
+// GetEvent return a channel which receives event
+func (pm *PodManager) GetEvent() chan *model.Message {
+	return pm.event
+}
+
+// AddPod adds pod in cache
+func (pm *PodManager) AddPod(k string, v types.FakePod) {
+	pm.pods.Store(k, v)
+}
+
+// DeletePod deletes pod in cache
+func (pm *PodManager) DeletePod(k string) {
+	pm.pods.Delete(k)
+}
+
+// UpdatePodStatus update pod status in cache
+func (pm *PodManager) UpdatePodStatus(k string, s string) {
+	v, ok := pm.pods.Load(k)
+	if ok {
+		pod := v.(types.FakePod)
+		pod.Status = s
+		pm.pods.Store(k, pod)
+	}
+}
+
+// GetPod gets pod from cache
+func (pm *PodManager) GetPod(key string) types.FakePod {
+	v, ok := pm.pods.Load(key)
+	if ok {
+		return v.(types.FakePod)
+	} else {
+		return types.FakePod{}
+	}
+}
+
+// ListPods lists all pods in cache
+func (pm *PodManager) ListPods() []types.FakePod {
+	pods := make([]types.FakePod, 0)
+	pm.pods.Range(func(k, v interface{}) bool {
+		pods = append(pods, v.(types.FakePod))
+		return true
+	})
+	return pods
+}
+
+// PodController is used to receive and process message
+func (pm *PodManager) PodController(w http.ResponseWriter, req *http.Request) {
+	switch req.Method {
+	case http.MethodGet:
+		// List Pod
+		log.LOGGER.Infof("Receive list pod request")
+		pods := pm.ListPods()
+		log.LOGGER.Infof("Current pods number: %v", len(pods))
+		rspBodyBytes := new(bytes.Buffer)
+		json.NewEncoder(rspBodyBytes).Encode(pods)
+		w.Write(rspBodyBytes.Bytes())
+		log.LOGGER.Infof("Finish list pod request")
+	case http.MethodPost:
+		log.LOGGER.Infof("Receive add pod request")
+		var p types.FakePod
+		// Get request body
+		if req.Body != nil {
+			body, err := ioutil.ReadAll(req.Body)
+			if err != nil {
+				log.LOGGER.Errorf("Read body error %v", err)
+				w.Write([]byte("Read request body error"))
+				return
+			}
+			log.LOGGER.Infof("Request body is %s", string(body))
```

Comment:
Please consider different log levels (v=2 , 3,4 etc , DEBUG etc) to log, since the tests will run for large scale pods.

Reply:
Thanks. will update

Reply:
Thanks @rohitsardesai83.
Currently these stubs are using
`github.com/kubeedge/beehive/pkg/common/log` as the same as the cloud and edge parts.
So we have some levels like `Fatal`,`Error`,`Warn`,`Info` and `Debug` to choose.
Based on your suggestion, some logs are adjusted into `Debug`.
So that we can reduce the logs when running the tests.
Thanks very much.